cloud-common/src/main/java/com/sequenceiq/cloudbreak/common/service/user/UserDetailsService.java

```diff
@@ -86,40 +88,83 @@ public IdentityUser getDetails(String username, UserFilterField filterField, Str
+        } catch (IOException e) {
+            throw new IllegalStateException(e);
+        }
+    }
+
     private IdentityUser getIdentityUser(String username, UserFilterField filterField, String clientSecret) {
         WebTarget target = getWebTarget(username, filterField);
         AccessToken accessToken = identityClient.getToken(clientSecret);
         String scimResponse = target.request(MediaType.APPLICATION_JSON).header("Authorization", "Bearer " + accessToken.getToken()).get(String.class);
         try {
             JsonNode root = JsonUtil.readTree(scimResponse);
-            List<IdentityUserRole> roles = new ArrayList<>();
-            String account = null;
             JsonNode userNode = root;
             if (UserFilterField.USERNAME.equals(filterField)) {
                 userNode = root.get("resources").get(0);
             }
             if (userNode == null) {
                 throw new UserDetailsUnavailableException("User details cannot be retrieved from identity server.");
             }
-            for (JsonNode node : userNode.get("groups")) {
-                String group = node.get("display").asText();
-                if (group.startsWith("sequenceiq.account")) {
-                    String[] parts = group.split("\\.");
-                    if (account != null && !account.equals(parts[ACCOUNT_PART])) {
-                        throw new IllegalStateException("A user can belong to only one account.");
-                    }
-                    account = parts[ACCOUNT_PART];
-                } else if (group.startsWith("sequenceiq.cloudbreak")) {
-                    String[] parts = group.split("\\.");
-                    roles.add(IdentityUserRole.fromString(parts[ROLE_PART]));
-                }
-            }
-            return createIdentityUser(roles, account, userNode);
+            return getIdentityUserFromJson(userNode);
         } catch (IOException e) {
             throw new UserDetailsUnavailableException("User details cannot be retrieved from identity server.", e);
         }
     }
 
+    private IdentityUser getIdentityUserFromJson(JsonNode userNode) {
+        String account = null;
+        List<IdentityUserRole> roles = new ArrayList<>();
+        for (JsonNode node : userNode.get("groups")) {
+            String group = node.get("display").asText();
+            if (group.startsWith("sequenceiq.account")) {
+                String[] parts = group.split("\\.");
+                if (account != null && !account.equals(parts[ACCOUNT_PART])) {
+                    throw new IllegalStateException("A user can belong to only one account.");
+                }
+                account = parts[ACCOUNT_PART];
+            } else if (group.startsWith("sequenceiq.cloudbreak")) {
+                String[] parts = group.split("\\.");
+                roles.add(IdentityUserRole.fromString(parts[ROLE_PART]));
+            }
+        }
+        return createIdentityUser(roles, account, userNode);
```

Comment:
Is account possible to pass value as null?

Reply:
Luckily no. Also it's an old and working code fragment